src/System.Drawing.Common/src/System/Drawing/BufferedGraphics.Unix.cs

```diff
@@ -28,42 +28,31 @@
 //
 //
 
-
 using System.ComponentModel;
 using System.Runtime.InteropServices;
 using System.Runtime.Serialization;
 
 namespace System.Drawing
 {
-    public sealed class BufferedGraphics : IDisposable
+    public sealed partial class BufferedGraphics
     {
         private Rectangle size;
         private Bitmap membmp = null;
-        private Graphics target = null;
         private Graphics source = null;
 
-        private BufferedGraphics()
-        {
-
-        }
-
-        internal BufferedGraphics(Graphics targetGraphics, Rectangle targetRectangle)
+        internal BufferedGraphics(Graphics targetGraphics, IntPtr targetDc, Rectangle targetRectangle)
         {
+            _targetGraphics = targetGraphics;
+            _targetDC = targetDc;
             size = targetRectangle;
-            target = targetGraphics;
             membmp = new Bitmap(size.Width, size.Height);
         }
 
-        ~BufferedGraphics()
-        {
-            Dispose(false);
-        }
-
         public Graphics Graphics
         {
             get
             {
-                if (source == null)
+                if (source == null && membmp != null)
```

Comment:
In what situation could membmp == null?

Reply:
When the graphics are disposed: https://github.com/hughbe/corefx/blob/9c6e31485652639887825f38a435953c79c62cc4/src/System.Drawing.Common/src/System/Drawing/BufferedGraphics.Unix.cs#L66-L70